server-common/src/test/java/org/apache/cassandra/sidecar/common/server/cluster/locator/TokenRangeTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.common.server.cluster.locator;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class TokenRangeTest
+{
+    @Test
+    void testEquals()
+    {
+        TokenRange r1 = new TokenRange(1, 100);
+        TokenRange r2 = new TokenRange(1, 100);
+        TokenRange r3 = new TokenRange(-10, 10);
+        assertThat(r1).isEqualTo(r2);
+        assertThat(r3).isNotEqualTo(r1)
+                      .isNotEqualTo(r2);
+    }
+
+    @Test
+    void testFirstToken()
+    {
+        TokenRange range = new TokenRange(1, 100);
+        assertThat(range.firstToken()).isEqualTo(Token.from(2));
+
+        TokenRange emptyRange = new TokenRange(1, 1);
+        assertThat(emptyRange.firstToken()).isNull();
+    }
+
+    @Test
+    void testCreateFromJavaDriverTokenRange()
+    {
+        com.datastax.driver.core.TokenRange ordinaryRange = mockRange(1L, 100L);
+        when(ordinaryRange.isWrappedAround()).thenReturn(false);
```

Comment:
I think isWrappedAround has an issue when the end range is the MIN_TOKEN: i.e. Murmur3, this would fail:

```
        com.datastax.driver.core.TokenRange range = mockRange(10L, Long.MIN_VALUE);
        when(range.isWrappedAround()).thenReturn(false);
         ranges = TokenRange.from(range);
        assertThat(ranges).hasSize(1)
                          .isEqualTo(Collections.singletonList(Range.openClosed(BigInteger.valueOf(10), Long.MAX_VALUE)));
```